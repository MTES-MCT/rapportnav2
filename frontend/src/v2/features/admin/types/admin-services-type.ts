export interface AdminService {
  id: number
  name: String
  createdAt?: Date
  updatedAt?: Date
  controlUnitId?: number
}

export type AdminAction = 'DELETE' | 'CREATE' | 'UPDATE'
export type AdminCell = { key: string; label: string; width: number }
