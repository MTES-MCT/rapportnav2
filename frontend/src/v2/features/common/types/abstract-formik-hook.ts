import { FormikErrors } from 'formik'
import { ObjectSchema } from 'yup'

export interface AbstractFormikHook<T, M> {
  initValue?: M
  errors?: FormikErrors<M>
  beforeInitValue: (value?: T) => M | undefined
  beforeSubmit: (value?: M, errors?: FormikErrors<M>) => T | undefined
  handleSubmit: (
    value?: M,
    errors?: FormikErrors<M>,
    onSubmit?: (valueToSubmit?: T) => Promise<unknown>
  ) => Promise<void>
}

export interface AbstractFormikSubFormHook<M> {
  initValue?: M
  errors?: FormikErrors<M>
  validationSchema?: ObjectSchema<any>
  handleSubmit: (value?: M, errors?: FormikErrors<M>) => Promise<void>
}
