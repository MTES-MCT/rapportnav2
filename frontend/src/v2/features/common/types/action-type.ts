export enum ActionType {
  CONTROL = 'CONTROL',
  NOTE = 'NOTE',
  SURVEILLANCE = 'SURVEILLANCE',
  STATUS = 'STATUS',
  CONTACT = 'CONTACT',
  RESCUE = 'RESCUE',
  OTHER = 'OTHER',
  NAUTICAL_EVENT = 'NAUTICAL_EVENT',
  VIGIMER = 'VIGIMER',
  ANTI_POLLUTION = 'ANTI_POLLUTION',
  BAAEM_PERMANENCE = 'BAAEM_PERMANENCE',
  PUBLIC_ORDER = 'PUBLIC_ORDER',
  REPRESENTATION = 'REPRESENTATION',
  ILLEGAL_IMMIGRATION = 'ILLEGAL_IMMIGRATION',
  INQUIRY = 'INQUIRY',
  COMMUNICATION = 'COMMUNICATION',
  TRAINING = 'TRAINING',
  UNIT_MANAGEMENT_PLANNING = 'UNIT_MANAGEMENT_PLANNING',
  UNIT_MANAGEMENT_TRAINING = 'UNIT_MANAGEMENT_TRAINING',
  UNIT_MANAGEMENT_OTHER = 'UNIT_MANAGEMENT_OTHER',
  CONTROL_SECTOR = 'CONTROL_SECTOR',
  CONTROL_NAUTICAL_LEISURE = 'CONTROL_NAUTICAL_LEISURE',
  CONTROL_SLEEPING_FISHING_GEAR = 'CONTROL_SLEEPING_FISHING_GEAR',
  OTHER_CONTROL = 'OTHER_CONTROL',
  RESOURCES_MAINTENANCE = 'RESOURCES_MAINTENANCE',
  MEETING = 'MEETING',
  PV_DRAFTING = 'PV_DRAFTING',
  HEARING_CONDUCT = 'HEARING_CONDUCT',
  LAND_SURVEILLANCE = 'LAND_SURVEILLANCE',
  FISHING_SURVEILLANCE = 'FISHING_SURVEILLANCE',
  MARITIME_SURVEILLANCE = 'MARITIME_SURVEILLANCE'
}

export enum ActionGroupType {
  OTHER_GROUP = 'OTHER_GROUP',
  CONTROL_GROUP = 'CONTROL_GROUP',
  SURVEILLANCE_GROUP = 'SURVEILLANCE_GROUP',
  ADMINISTRATIVE_GROUP = 'ADMINISTRATIVE_GROUP',
  UNIT_MANAGEMENT_GROUP = 'UNIT_MANAGEMENT_GROUP'
}
