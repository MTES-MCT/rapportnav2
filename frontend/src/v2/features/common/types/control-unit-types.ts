export type ControlUnitResource = {
  id: string
  controlUnit: ControlUnit
  controlUnitId: number
  isArchived: boolean
  name: string
  note?: string
  photo?: string
  station: Station
  stationId: number
  type: ControlUnitResourceType
}

export type Station = {
  id: number
  latitude: number
  longitude: number
  name: string
}

export type ControlUnit = {
  id: number
  areaNote?: string
  administrationId: number
  departmentAreaInseeCode?: string
  isArchived: boolean
  name: string
  termsNote?: string
}

export type Administration = {
  id: number
  name: string
  controlUnitIds: number[]
  isArchived: boolean
  controlUnits: ControlUnit[]
}

enum ControlUnitResourceType {
  AIRPLANE = "AIRPLANE",
  BARGE = "BARGE",
  CAR = "CAR",
  DRONE = "DRONE",
  EQUESTRIAN = "EQUESTRIAN",
  FAST_BOAT = "FAST_BOAT",
  FRIGATE = "FRIGATE",
  HELICOPTER = "HELICOPTER",
  HYDROGRAPHIC_SHIP = "HYDROGRAPHIC_SHIP",
  KAYAK = "KAYAK",
  LIGHT_FAST_BOAT = "LIGHT_FAST_BOAT",
  MINE_DIVER = "MINE_DIVER",
  MOTORCYCLE = "MOTORCYCLE",
  NET_LIFTER = "NET_LIFTER",
  NO_RESOURCE = "NO_RESOURCE",
  OTHER = "OTHER",
  PATROL_BOAT = "PATROL_BOAT",
  PEDESTRIAN = "PEDESTRIAN",
  PIROGUE = "PIROGUE",
  RIGID_HULL = "RIGID_HULL",
  SEA_SCOOTER = "SEA_SCOOTER",
  SEMI_RIGID = "SEMI_RIGID",
  SUPPORT_SHIP = "SUPPORT_SHIP",
  TRAINING_SHIP = "TRAINING_SHIP",
  TUGBOAT = "TUGBOAT"
}
