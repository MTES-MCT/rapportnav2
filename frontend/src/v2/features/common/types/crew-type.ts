import { Service } from '@common/types/crew-types.ts'

export type Agent = {
  id: number
  firstName: string
  lastName: string
  services: Service[]
}

export type AgentRole = {
  id: string
  title: string
}

export type MissionCrewMember = {
  id?: string
  agent: Agent
  comment?: string
  role?: AgentRole
  missionId?: number
}
