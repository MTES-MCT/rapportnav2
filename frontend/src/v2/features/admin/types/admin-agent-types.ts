import { AdminService } from './admin-services-type'

export type Agent = {
  id?: string
  firstName?: string
  lastName?: string
  services?: string[]
  deletedAt?: Date
  createdAt?: Date
  updatedAt?: Date
}

export type AgentRole = {
  id: string
  title: string
}

export type AdminAgentService = {
  id: number
  agent: Agent
  role: AgentRole
  createdAt?: Date
  updatedAt?: Date
  disabledAt?: Date
}

export type AdminServiceWithAgent = {
  service: AdminService
  agentServices: AdminAgentService[]
}

export type AdminAgentServiceInput = {
  id?: number
  agentId: number
  serviceId: number
  roleId?: number
}
