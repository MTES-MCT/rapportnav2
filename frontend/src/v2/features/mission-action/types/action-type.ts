import { MissionEnvActionData, MissionFishActionData, MissionNavActionData } from '../../common/types/mission-action'
import { MissionActionData } from '../../common/types/mission-action-data'
import { RescueType } from '../../common/types/rescue-type'

export type ActionGenericDateObservation = {
  id: string
  startDateTimeUtc: string
  endDateTimeUtc: string
  observations: string
}

export type ActionGenericDateObservationInput = {
  dates: Date[]
} & MissionActionData

export type ActionAntiPollutionInput = {
  dates: Date[]
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionIllegalImmigrationInput = {
  dates: Date[]
  isMissionFinished?: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionFreeNoteInput = {
  date: Date
} & MissionNavActionData

export type ActionRescueInput = {
  dates: Date[]
  rescueType: RescueType
  isMissionFinished?: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionStatusInput = {
  date: Date
} & MissionNavActionData

export type ActionSurveillanceInput = {
  dates: Date[]
} & MissionEnvActionData

export type ActionNavControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionFishControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: (number | undefined)[]
} & MissionFishActionData

export type ActionEnvControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: [number?, number?]
} & MissionEnvActionData
