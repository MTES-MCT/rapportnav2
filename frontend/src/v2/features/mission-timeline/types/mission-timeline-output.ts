import { ActionStatusReason, ActionStatusType } from '@common/types/action-types'
import { ControlMethod, ControlType } from '@common/types/control-types'
import {
  ActionTargetTypeEnum,
  ActionTypeEnum,
  MissionSourceEnum,
  VehicleTypeEnum,
  VesselSizeEnum,
  VesselTypeEnum
} from '@common/types/env-mission-types'
import { MissionActionType } from '@common/types/fish-mission-types'
import { CompletenessForStats } from '@common/types/mission-types'

export type MissionTimelineAction = {
  id: string
  type: ActionTypeEnum
  missionId: number
  source?: MissionSourceEnum
  status?: ActionStatusType
  fishActionType?: MissionActionType
  isCompleteForStats?: boolean
  summaryTags?: string[]
  completenessForStats?: CompletenessForStats
  startDateTimeUtc?: string
  endDateTimeUtc?: string
  vesselId?: number
  vesselName?: string
  actionNumberOfControls?: number
  actionTargetType?: ActionTargetTypeEnum
  vehicleType?: VehicleTypeEnum
  controlsToComplete?: ControlType[]
  formattedControlPlans?: any[]
  observations?: string
  isPersonRescue?: Boolean
  isVesselRescue?: Boolean
  reason?: ActionStatusReason
  controlMethod?: ControlMethod
  vesselIdentifier?: string
  vesselType?: VesselTypeEnum
  vesselSize?: VesselSizeEnum
}
