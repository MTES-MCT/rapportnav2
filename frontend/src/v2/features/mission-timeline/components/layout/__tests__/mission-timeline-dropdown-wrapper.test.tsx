import { Icon } from '@mtes-mct/monitor-ui'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { ActionGroupType, ActionType } from '../../../../common/types/action-type'
import MissionTimelineDropdownWrapper from '../mission-timeline-dropdown-wrapper'

const dropdownItems = [
  {
    type: ActionGroupType.CONTROL_GROUP,
    icon: Icon.ControlUnit,
    dropdownText: 'Ajouter des contrôles',
    subItems: [{ type: ActionType.CONTROL, dropdownText: 'Contrôle de navire' }]
  }
]

const handleSelect = vi.fn()

describe('MissionTimelineDropdownWrapper', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<MissionTimelineDropdownWrapper dropdownItems={dropdownItems} onSelect={handleSelect} />)
    expect(wrapper).toMatchSnapshot()
  })
})
