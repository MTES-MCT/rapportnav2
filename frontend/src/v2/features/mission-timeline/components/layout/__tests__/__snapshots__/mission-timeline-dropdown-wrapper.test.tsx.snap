// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionTimelineDropdownWrapper > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jMsorb bqjWtF Component-Dropdow rs-dropdown rs-dropdown-placement-bottom-end"
      >
        <button
          aria-controls="menu-:r1:"
          aria-haspopup="menu"
          class="rs-dropdown-toggle rs-btn rs-btn-default"
          id="menubutton-:r0:"
          role="button"
          type="button"
        >
          <span
            class="sc-gDpztx bVQkEh Element-IconBox"
          >
            <svg
              class="rs-dropdown-toggle-icon"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <g
                transform="translate(120 -82)"
              >
                <path
                  d="M-102,91h-7V84h-2v7h-7v2h7v7h2V93h7Z"
                  fill="currentColor"
                />
                <rect
                  fill="none"
                  height="20"
                  transform="translate(-120 82)"
                  width="20"
                />
              </g>
            </svg>
          </span>
          Ajouter
          <svg
            aria-hidden="true"
            aria-label="angle down"
            class="rs-dropdown-toggle-caret rs-icon"
            data-category=""
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 21 32"
            width="1em"
          >
            <path
              d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
            />
          </svg>
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </button>
        <ul
          aria-labelledby="menubutton-:r0:"
          class="sc-jMsorb bqjWtF Component-Dropdow rs-dropdown-menu"
          hidden=""
          id="menu-:r1:"
          role="menu"
          tabindex="0"
        >
          <div>
            <li
              aria-disabled="false"
              class="sc-icnseD iChHAy rs-dropdown-item rs-dropdown-item-with-icon"
              data-event-key="CONTROL_GROUP"
              id="menuitem-:r2:"
              role="menuitem"
              style="min-width: 300px;"
              tabindex="-1"
            >
              <span
                class="sc-gDpztx bVQkEh Element-IconBox"
              >
                <svg
                  class="rs-dropdown-item-menu-icon"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <g
                    transform="translate(80 205)"
                  >
                    <rect
                      fill="currentColor"
                      height="12"
                      transform="translate(-64 -188) rotate(90)"
                      width="2"
                    />
                    <path
                      d="M-67.694-199l1.333,8h-7.277l1.336-8Zm0-2h-4.6a2,2,0,0,0-1.977,1.675L-76-189h12l-1.721-10.325A2,2,0,0,0-67.7-201Z"
                      fill="currentColor"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-70.75 -204.25)"
                      width="1.5"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-64.061 -203) rotate(45)"
                      width="1.5"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-77 -201.939) rotate(-45)"
                      width="1.5"
                    />
                  </g>
                  <rect
                    fill="none"
                    height="20"
                    width="20"
                  />
                </svg>
              </span>
              Ajouter des contrôles
            </li>
          </div>
        </ul>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jMsorb bqjWtF Component-Dropdow rs-dropdown rs-dropdown-placement-bottom-end"
    >
      <button
        aria-controls="menu-:r1:"
        aria-haspopup="menu"
        class="rs-dropdown-toggle rs-btn rs-btn-default"
        id="menubutton-:r0:"
        role="button"
        type="button"
      >
        <span
          class="sc-gDpztx bVQkEh Element-IconBox"
        >
          <svg
            class="rs-dropdown-toggle-icon"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <g
              transform="translate(120 -82)"
            >
              <path
                d="M-102,91h-7V84h-2v7h-7v2h7v7h2V93h7Z"
                fill="currentColor"
              />
              <rect
                fill="none"
                height="20"
                transform="translate(-120 82)"
                width="20"
              />
            </g>
          </svg>
        </span>
        Ajouter
        <svg
          aria-hidden="true"
          aria-label="angle down"
          class="rs-dropdown-toggle-caret rs-icon"
          data-category=""
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 21 32"
          width="1em"
        >
          <path
            d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
          />
        </svg>
        <span
          class="rs-ripple-pond"
        >
          <span
            class="rs-ripple"
          />
        </span>
      </button>
      <ul
        aria-labelledby="menubutton-:r0:"
        class="sc-jMsorb bqjWtF Component-Dropdow rs-dropdown-menu"
        hidden=""
        id="menu-:r1:"
        role="menu"
        tabindex="0"
      >
        <div>
          <li
            aria-disabled="false"
            class="sc-icnseD iChHAy rs-dropdown-item rs-dropdown-item-with-icon"
            data-event-key="CONTROL_GROUP"
            id="menuitem-:r2:"
            role="menuitem"
            style="min-width: 300px;"
            tabindex="-1"
          >
            <span
              class="sc-gDpztx bVQkEh Element-IconBox"
            >
              <svg
                class="rs-dropdown-item-menu-icon"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <g
                  transform="translate(80 205)"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    transform="translate(-64 -188) rotate(90)"
                    width="2"
                  />
                  <path
                    d="M-67.694-199l1.333,8h-7.277l1.336-8Zm0-2h-4.6a2,2,0,0,0-1.977,1.675L-76-189h12l-1.721-10.325A2,2,0,0,0-67.7-201Z"
                    fill="currentColor"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-70.75 -204.25)"
                    width="1.5"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-64.061 -203) rotate(45)"
                    width="1.5"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-77 -201.939) rotate(-45)"
                    width="1.5"
                  />
                </g>
                <rect
                  fill="none"
                  height="20"
                  width="20"
                />
              </svg>
            </span>
            Ajouter des contrôles
          </li>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
